test/lib/utils.js

```diff
@@ -52,7 +53,17 @@ function doDump(object, trailer) {
   dump(JSON.stringify(object, null, 1) + trailer); // eslint-disable-line no-undef
 }
 
+function mockActivityStream(options = {}) {
```

Comment:
We shouldn't call this 'mockActivityStream', because that signals to the caller that we're returning a fake ActivityStream that doesn't implement any real behaviour, which isn't true, it's a real instance of ActivityStream just with a mocked out MetadataStore.  So we should call this getActivityStream or getTestActivityStream to indicate that it's a real ActivityStream, but should only be used for test purposes, not real code.  Other than that this looks great!